core/services/job/orm.go

```diff
@@ -252,18 +253,60 @@ func (o *orm) CreateJob(jb *Job, qopts ...pg.QOpt) error {
 					if !useForwarders && err != nil {
 						return errors.Wrapf(ErrNoSuchTransmitterKey, "%v", jb.OCR2OracleSpec.TransmitterID)
 					}
+
+					chainIDInterface, ok := jb.OCR2OracleSpec.RelayConfig["chainID"]
+					if !ok {
+						return errors.Errorf("missing chainID in EVM type in OCR2 job spec")
+					}
+					newChainID, ok := chainIDInterface.(int64)
+					if !ok {
+						return errors.Errorf("failed to parse chainID in OCR2 job spec")
+					}
+
+					var specs []OCR2OracleSpec
+					err = tx.Select(&specs, `SELECT * FROM ocr2_oracle_specs WHERE relay = $1 AND contract_id = $2`,
+						"EVM", jb.OCR2OracleSpec.ContractID,
```

Comment:
I'm worried about leaving this loop around. Even if we only expect a few results for now, this could easily become hundreds or thousands for some new use case. Are we able to use the postgres json queries? Something like this:
```suggestion
					err = tx.Select(&specs, `SELECT * FROM ocr2_oracle_specs WHERE relay = $1 AND contract_id = $2 AND relay_config ->> 'chainID' = $3`,
						"EVM", jb.OCR2OracleSpec.ContractID, 
```
There is a `#>>` command that takes a path to the field as well, if we wanted to be a little more generic about it, and support other chain types: `relay_config #>> '{path,to,chainID}'`
https://www.postgresql.org/docs/current/functions-json.html